opslevel/resource_opslevel_property_assignment.go

```diff
@@ -1,113 +1,175 @@
-
-// 	return resourcePropertyAssignmentRead(d, client)
-// }
-
-// func resourcePropertyAssignmentRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	// an invalid id can be passed in by using 'terraform import', validate the it before attaching the id to the resource
-// 	parts := strings.SplitN(d.Id(), ":", 2)
-// 	if len(parts) != 2 {
-// 		return fmt.Errorf("[%s] invalid property assignment id, should be in format 'ownerId:definitionId' (only a single colon between both ids, no spaces or special characters)", d.Id())
-// 	}
-// 	ownerId := parts[0]
-// 	definitionId := parts[1]
-// 	if !opslevel.IsID(ownerId) {
-// 		return fmt.Errorf("[%s] invalid ownerId", ownerId)
-// 	}
-// 	if !opslevel.IsID(definitionId) {
-// 		return fmt.Errorf("[%s] invalid definitionId", definitionId)
-// 	}
-
-// 	resource, err := client.GetProperty(ownerId, definitionId)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	// if resource was fetched correctly, attach the id to the resource
-// 	d.SetId(fmt.Sprintf("%s:%s", ownerId, definitionId))
-
-// 	if err := d.Set("definition", d.Get("definition")); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("owner", d.Get("owner")); err != nil {
-// 		return err
-// 	}
-// 	if err := d.Set("value", string(*resource.Value)); err != nil {
-// 		return err
-// 	}
-
-// 	return nil
-// }
-
-// func resourcePropertyAssignmentDelete(d *schema.ResourceData, client *opslevel.Client) error {
-// 	id := strings.Split(d.Id(), ":")
-// 	if len(id) != 2 {
-// 		return fmt.Errorf("[%s] invalid property assignment id, should be in format 'ownerId:definitionId' (only a single colon between both ids, no spaces or special characters)", d.Id())
-// 	}
-// 	ownerId := id[0]
-// 	definitionId := id[1]
-// 	err := client.PropertyUnassign(ownerId, definitionId)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.SetId("")
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &PropertyAssignmentResource{}
+
+type PropertyAssignmentResource struct {
+	CommonResourceClient
+}
+
+func NewPropertyAssignmentResource() resource.Resource {
+	return &PropertyAssignmentResource{}
+}
+
+type PropertyAssignmentResourceModel struct {
+	Definition  types.String `tfsdk:"definition"`
+	Id          types.String `tfsdk:"id"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+	Locked      types.Bool   `tfsdk:"locked"`
```

Comment:
`Locked` isn't used in the create or read operations. Is this field needed?

Reply:
i cannot think of a way that will ever be true except if for some reason some add a property to a service via Terraform and then moves / copies it into opslevel.yaml and then things get really screwed up.  But either way it should blow up with API errors then.

fine to leave in IMO